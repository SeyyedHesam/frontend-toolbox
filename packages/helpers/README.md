# Helper methods
> ðŸ˜Ž it is useful to have mini helper functions which helps you with some thing
----

[![version](https://img.shields.io/npm/v/@snappmarket/helpers.svg?style=flat-square)](https://www.npmjs.com/package/@snappmarket/helpers)
[![downloads](https://img.shields.io/npm/dm/@snappmarket/helpers.svg?style=flat-square)](http://www.npmtrends.com/@snappmarket/helpers)
[![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square)](http://makeapullrequest.com)
[![MIT License](https://img.shields.io/npm/l/@snappmarket/helpers.svg?style=flat-square)](https://github.com/snappmarket/react-hooks/tree/master/packages/useDidUpdateEffect/blob/master/LICENSE.md)

[![Watch on GitHub](https://img.shields.io/github/watchers/snappmarket/react-hooks.svg?style=social)](https://github.com/snappmarket/react-hooks/watchers)
[![Star on GitHub](https://img.shields.io/github/stars/snappmarket/react-hooks.svg?style=social)](https://github.com/snappmarket/react-hooks/stargazers)


## Helpers Found Here

| Helper                                                                                                                | Methods |
| --------------------------------------------------------------------------------------------------------------------- | -------- | 
| [ArrayHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/ArrayHelpers)         | `arrayItemAddProp` ,f`lattenArray` ,s`erializeObject` ,s`tringifyArray` ,d`eepFlatten`   |
| [CookieHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/CookieHelpers)       | `getCookie`, `setCookie`   |  
| [DebugHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/DebugHelpers)         | `delog`, `ApiError`, `globalErrorCatcher`    |  
| [HttpHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/HttpHelpers)           | `fetchWithTimeOut`, `universalCall`   |  
| [JsonHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/JsonHelpers)           | `recursiveJsonParse` |  
| [MapHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/MapHelpers)             | `isLocationInIran`    | 
| [NumberHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/NumberHelpers)       | `persianNumber`, `englishNumber`, `generateKey`, `cellphoneValidate`, `cellphoneValidatePattern`, `currencyPrice`  | 
| [ObjectHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/ObjectHelpers)       | `removeByKey`, `flattenObject`, `makeCookieString`, `safeObjectPropertyRead`, `getNextProp`   | 
| [OSHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/OSHelpers)               | `getOSName`, `getAgent` | 
| [PromiseHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/PromiseHelpers)     | `racePromise`, `makeTimeout`, `emptyPromise`  | 
| [ProxyHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/ProxyHelpers)         | `onChangeReflector` | 
| [ReduxHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/ReduxHelpers)         | `actionMaker`, `normalizeActionType`, `configPersistor`   | 
| [StorageHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/StorageHelpers)     | `getLocalStorageData` |  
| [StringHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/StringHelpers)       | `stringReplace`, `bulkStringReplace` | 
| [TemplateHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/TemplateHelpers)   | `HexToRgb`, `defineForegroundColor`, `makeRgba`, `decideMeasurement`, `makeShadow`  | 
| [WindowHelpers](https://github.com/snappmarket/react-hooks/tree/master/packages/helpers/packages/WindowHelpers)       | `getPathName`, `getWindowSize`    | 

