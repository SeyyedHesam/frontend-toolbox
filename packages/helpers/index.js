export * from './src/arrayHelpers';
export * from './src/cookieHelper';
export * from './src/debugHelper';
export * from './src/httpHelpers';
export * from './src/jsonHelpers';
export * from './src/localStorageHelpers';
export * from './src/mapHelpers';
export * from './src/numberHelpers';
export * from './src/objectHelper';
export * from './src/osHelpers';
export * from './src/productHelpers';
export * from './src/promiseHelpers';
export * from './src/proxyHelpers';
export * from './src/reduxHelper';
export * from './src/stringHelpers';
export * from './src/themeHelpers';
export * from './src/arrayHelpers';
