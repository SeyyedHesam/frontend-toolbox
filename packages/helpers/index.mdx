---
name: .intro
route: /helpers
menu: Helpers
---

# âš™ SnappMarket frontend powerful helpers
We provide suitable solution to got none opinionated toolbox


## get started
```bash
npm i @snappmarket/helpers

// or

yarn add @snappmarket/helpers
```


## ðŸ’Š Powerful helpers instead of lodash

| Helper                                        | Methods                                                                                                          |
| ----------------------------------------------| ---------------------------------------------------------------------------------------------------------------- |                                                                                                         |
| [ArrayHelpers](/helpers/ArrayHelpers)         | `arrayItemAddProp` ,f`lattenArray` ,s`erializeObject` ,s`tringifyArray` ,d`eepFlatten`                           |
| [CookieHelpers](/helpers/CookieHelpers)       | `getCookie`, `setCookie`                                                                                         |
| [DebugHelpers](/helpers/DebugHelpers)         | `delog`, `ApiError`, `globalErrorCatcher`                                                                        |
| [HttpHelpers](/helpers/HttpHelpers)           | `fetchWithTimeOut`, `universalCall`                                                                              |
| [JsonHelpers](/helpers/JsonHelpers)           | `recursiveJsonParse`                                                                                             |
| [MapHelpers](/helpers/MapHelpers)             | `isLocationInIran`                                                                                               |
| [NumberHelpers](/helpers/NumberHelpers)       | `persianNumber`, `englishNumber`, `generateKey`, `cellphoneValidate`, `cellphoneValidatePattern`, `currencyPrice`|
| [ObjectHelpers](/helpers/ObjectHelpers)       | `removeByKey`, `flattenObject`, `makeCookieString`, `safeObjectPropertyRead`, `getNextProp`                      |
| [OSHelpers](/helpers/OSHelpers)               | `getOSName`, `getAgent`                                                                                          |
| [PromiseHelpers](/helpers/PromiseHelpers)     | `racePromise`, `makeTimeout`, `emptyPromise`                                                                     |
| [ProxyHelpers](/helpers/ProxyHelpers)         | `onChangeReflector`                                                                                              |
| [ReduxHelpers](/helpers/ReduxHelpers)         | `actionMaker`, `normalizeActionType`, `configPersistor`                                                          |
| [StorageHelpers](/helpers/StorageHelpers)     | `getLocalStorageData`                                                                                            |
| [StringHelpers](/helpers/StringHelpers)       | `stringReplace`, `bulkStringReplace`                                                                             |
| [TemplateHelpers](/helpers/TemplateHelpers)   | `HexToRgb`, `defineForegroundColor`, `makeRgba`, `decideMeasurement`, `makeShadow`                               |
| [WindowHelpers](/helpers/WindowHelpers)       | `getPathName`, `getWindowSize`                                                                                   |

