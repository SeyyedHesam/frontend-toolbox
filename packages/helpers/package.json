{
  "name": "@snappmarket/helpers",
  "private": false,
  "version": "0.1.26",
  "author": "Jafar Rezaei (SayJeyHi) <jafar.rezaei.ard@gmail.com>",
  "main": "dist/index.js",
  "module": "dist/index.es.js",
  "scripts": {
    "test": "cross-env CI=1 react-scripts test --env=jsdom",
    "test:watch": "react-scripts test --env=jsdom",
    "build": "rollup -c",
    "start": "rollup -c -w",
    "prepare": "yarn run build",
    "predeploy": "cd example && yarn install && yarn run build",
    "deploy": "gh-pages -d example/build"
  },
  "dependencies": {
    "isomorphic-unfetch": "^3.0.0"
  },
  "peerDependencies": {
    "react": "^16.8.6"
  },
  "engines": {
    "node": ">=8",
    "npm": ">=5"
  },
  "license": "MIT",
  "keywords": [
    "helpers",
    "custom-functions",
    "loadash",
    "low-size-lodash",
    "array-helpers",
    "cookie-helpers"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/snappmarket/frontend-toolbox"
  },
  "gitHead": "2207934c0e1e69a5004aee326f2ccfabe910ee32"
}
