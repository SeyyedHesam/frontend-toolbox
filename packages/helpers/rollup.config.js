import babel from 'rollup-plugin-babel';
import commonjs from 'rollup-plugin-commonjs';
import external from 'rollup-plugin-peer-deps-external';
import resolve from 'rollup-plugin-node-resolve';
import url from 'rollup-plugin-url';
import builtins from 'rollup-plugin-node-builtins';
import globals from 'rollup-plugin-node-globals';

import pkg from './package.json';

export default {
  input : 'index.js',
  output: [
    {
      file     : pkg.main,
      format   : 'cjs',
      exports  : 'named',
      sourcemap: true
    },
    {
      file     : pkg.module,
      format   : 'es',
      exports  : 'named',
      sourcemap: true
    }
  ],
  plugins: [
    globals(),
    builtins(),
    external(),
    url({ exclude: ['**/*.svg'] }),
    babel({
      exclude: 'node_modules/**'
    }),
    resolve(),
    commonjs({
      include: 'node_modules/**',
      namedExports: {
        'node_modules/react-is/index.js': ['isValidElementType']
      }
    })
  ]
};
