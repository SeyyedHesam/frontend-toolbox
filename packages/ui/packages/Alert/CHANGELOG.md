# @snappmarket/ui-alert

## 0.2.0
### Minor Changes

- 4bf49e8: First minor package publish with change-set

### Patch Changes

- Updated dependencies [4bf49e8]
  - @snappmarket/icons@0.2.0
