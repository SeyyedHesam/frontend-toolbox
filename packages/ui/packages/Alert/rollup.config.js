import config from '../../rollup.config';

export default {
  ...config[0],
  input: 'index.js'
};
