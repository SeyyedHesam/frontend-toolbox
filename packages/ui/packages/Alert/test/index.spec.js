import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { Wrapper, theme } from '../../../test/test.helpers';
import Alert from '../index';


describe('Alert ui component tests', () => {
  it('Should render with orange color at default', () => {
    const { queryByTestId } = render(
      <Wrapper><Alert data-testid="alert-default-value"><h3>some text !</h3></Alert></Wrapper>,
    );

    expect(queryByTestId('alert-default-value')).toHaveStyle({
      backgroundColor: theme.colors.orange.bright,
      color: theme.colors.orange.dark,
    });
  });

  it('Should changes the class when get different props', () => {
    const { queryByTestId, rerender } = render(
      <Wrapper><Alert data-testid="alert" status="error"><h3>some text !</h3></Alert></Wrapper>,
    );

    expect(queryByTestId('alert')).toHaveStyle({
      backgroundColor: theme.colors.red.bright,
      color: theme.colors.red.dark,
    });

    rerender(<Wrapper><Alert data-testid="alert" status="info"><h3>some text !</h3></Alert></Wrapper>);
    expect(queryByTestId('alert')).toHaveStyle({
      backgroundColor: theme.colors.blue.bright,
      color: theme.colors.blue.dark,
    });

    rerender(<Wrapper><Alert data-testid="alert" status="warning"><h3>some text !</h3></Alert></Wrapper>);
    expect(queryByTestId('alert')).toHaveStyle({
      backgroundColor: theme.colors.yellow.bright,
      color: theme.colors.yellow.dark,
    });

    rerender(<Wrapper><Alert data-testid="alert" status="success"><h3>some text !</h3></Alert></Wrapper>);
    expect(queryByTestId('alert')).toHaveStyle({
      backgroundColor: theme.colors.green.bright,
      color: theme.colors.green.dark,
    });
  });

  it('Should close Alert on close button click', () => {
    const { queryByTestId } = render(
      <Wrapper><Alert data-testid="alert" status="error" closable="true"><h3>some text !</h3></Alert></Wrapper>,
    );

    fireEvent(queryByTestId('close-button'), 'click');
    expect(queryByTestId('alert')).not.toBeVisible();
  });
});
