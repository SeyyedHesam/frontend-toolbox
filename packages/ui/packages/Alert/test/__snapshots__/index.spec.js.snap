// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert tests Should changes the class when get different props 1`] = `
<div
  className="sc-EHOje dFtRRb"
  color="red"
>
  <button
    className="sc-bdVaJa sc-bZQynM fTnRNf circle d-inline-block pull-left closeBtn"
    color="gray"
    onClick={[Function]}
    size="xs"
  />
  <h3>
    مشکلی وجود دارد
  </h3>
</div>
`;

exports[`Alert tests Should changes the class when get different props 2`] = `
<div
  className="sc-EHOje dFtRRb"
  color="red"
>
  <button
    className="sc-bdVaJa sc-bZQynM fTnRNf circle d-inline-block pull-left closeBtn"
    color="gray"
    onClick={[Function]}
    size="xs"
  />
  <h3>
    مشکلی وجود دارد
  </h3>
</div>
`;

exports[`Alert tests Should changes the class when get different props 3`] = `
<div
  className="sc-EHOje dFtRRb"
  color="red"
>
  <button
    className="sc-bdVaJa sc-bZQynM fTnRNf circle d-inline-block pull-left closeBtn"
    color="gray"
    onClick={[Function]}
    size="xs"
  />
  <h3>
    مشکلی وجود دارد
  </h3>
</div>
`;

exports[`Alert tests Should changes the class when get different props 4`] = `
<div
  className="sc-EHOje dFtRRb"
  color="red"
>
  <button
    className="sc-bdVaJa sc-bZQynM fTnRNf circle d-inline-block pull-left closeBtn"
    color="gray"
    onClick={[Function]}
    size="xs"
  />
  <h3>
    مشکلی وجود دارد
  </h3>
</div>
`;

exports[`Alert tests Should changes the class when get different props 5`] = `
<div
  className="sc-EHOje dFtRRb"
  color="red"
>
  <button
    className="sc-bdVaJa sc-bZQynM fTnRNf circle d-inline-block pull-left closeBtn"
    color="gray"
    onClick={[Function]}
    size="xs"
  />
  <h3>
    مشکلی وجود دارد
  </h3>
</div>
`;
